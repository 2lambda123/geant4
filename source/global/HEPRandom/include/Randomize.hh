// This code implementation is the intellectual property of
// the GEANT4 collaboration.
//
// By copying, distributing or modifying the Program (or any work
// based on the Program) you indicate your acceptance of this statement,
// and all its terms.
//
// $Id: Randomize.hh,v 1.3 2000-08-02 10:48:21 gcosmo Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
#ifndef randomize_h
#define randomize_h 1

#include <CLHEP/Random/Randomize.h>

// Additional distributions ...
//
#include <CLHEP/Random/RandGaussQ.h>
#include <CLHEP/Random/RandGaussT.h>
#include <CLHEP/Random/RandPoissonQ.h>
#include <CLHEP/Random/RandPoissonT.h>
#include <CLHEP/Random/RandLandau.h>
#include <CLHEP/Random/RandBit.h>

#define G4UniformRand() HepRandom::getTheEngine()->flat()

#endif // randomize_h 
