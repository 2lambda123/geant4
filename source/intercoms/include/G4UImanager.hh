//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: G4UImanager.hh,v 1.12 2001-10-02 00:32:06 asaim Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef G4UImanager_h
#define G4UImanager_h 1

#include "globals.hh"
#include "g4rw/ctoken.h"
//#include "g4rw/tvordvec.h"
#include "g4std/vector"
#include "g4std/fstream"
#include "G4VStateDependent.hh"
#include "G4UIcommandStatus.hh"
class G4UIcommandTree;
class G4UIcommand;
class G4UIsession;
class G4UIcontrolMessenger;
class G4UnitsMessenger;
class G4UIaliasList;

// class description:
//
//  This is a singlton class which controls the command manipulation
// and the user interface(s). The constructor of this class MUST NOT 
// invoked by the user.
//

class G4UImanager : public G4VStateDependent
{
  public: // with description
      static G4UImanager * GetUIpointer();
      //  A static method to get the pointer to the only existing object
      // of this class.

  protected:
      G4UImanager();
  public:
      ~G4UImanager();
  private:
      G4UImanager(const G4UImanager &right);
      const G4UImanager & operator=(const G4UImanager &right);
      int operator==(const G4UImanager &right) const;
      int operator!=(const G4UImanager &right) const;

  public: // with description
      G4String GetCurrentValues(const char * aCommand);
      //  This method returns a string which represents the current value(s)
      // of the parameter(s) of the specified command. Null string will be
      // returned if the given command is not defined or the command does
      // not support the GetCurrentValues() method.
      void AddNewCommand(G4UIcommand * newCommand);
      //  This method register a new command.
      void RemoveCommand(G4UIcommand * aCommand);
      //  This command remove the registered command. After invokation of this
      // command, that particular command cannot be applied.
      void ExecuteMacroFile(G4String fileName);
      //  A macro file defined by the argument will be read by G4UIbatch object.
      G4int ApplyCommand(const char * aCommand);
      G4int ApplyCommand(G4String aCommand);
      //  These two methods are identical. A command (and parameter(s)) given
      // by the method's argument will be applied. Zero will be returned in 
      // case the command is successfully executed. Positive non-zero value
      // will be returned if the command couldn't be executed. The meaning of
      // this non-zero value is the following.
      //   The returned number : xyy
      //        x00 : G4CommandStatus.hh enumeration
      //         yy : the problematic parameter (first found)
      void StoreHistory(G4String fileName="G4history.macro");
      void StoreHistory(G4bool historySwitch,
             G4String fileName="G4history.macro");
      //  The executed commands will be stored in the defined file. If 
      // "historySwitch" is false, saving will be suspended.
      void ListCommands(G4String direc);
      //  All commands registored under the given directory will be listed to
      // G4cout.
      void SetAlias(const char * aliasLine);
      void SetAlias(G4String aliasLine);
      //  Define an alias. The first word of "aliasLine" string is the
      // alias name and the remaining word(s) is(are) string value
      // to be aliased.
      void RemoveAlias(const char * aliasName);
      void RemoveAlias(G4String aliasName);
      //  Remove the defined alias.
      void ListAlias();
      //  Print all aliases.


  public:
      //void Foreach(G4String macroFile,G4String variableName,
      //             G4double initialValue,G4double finalValue,G4double stepSize);
      //void Foreach(G4String macroFile,G4String variableName,
      //             G4int initialValue,G4int finalValue,G4int stepSize);

      virtual G4bool Notify(G4ApplicationState requestedState);
      //  This method is exclusively invoked by G4StateManager and the user
      // must not use this method.
 
  private:
      void PauseSession(G4String msg);
      void CreateMessenger();
      G4UIcommandTree* FindDirectory(const char* dirName);

  public:
  // following three methods will be removed quite soon.
      void Interact();
      void Interact(const char * promptCharacters);
      void Interact(G4String promptCharacters);

  private:
      static G4UImanager * fUImanager;
      static G4bool fUImanagerHasBeenKilled;
      G4UIcommandTree * treeTop;
      G4UIsession * session;
      G4UIcontrolMessenger * UImessenger;
      G4UnitsMessenger * UnitsMessenger;
      G4String savedParameters;
      G4UIcommand * savedCommand;
      G4int verboseLevel;
      G4std::ofstream historyFile;
      G4bool saveHistory;
      G4std::vector<G4String> histVec;
      G4UIaliasList* aliasList;
      
      G4bool pauseAtBeginOfEvent;
      G4bool pauseAtEndOfEvent;


  public: // with description
      G4String GetCurrentStringValue(const char * aCommand, 
	    int parameterNumber=1, G4bool reGet=true);
      G4int GetCurrentIntValue(const char * aCommand, 
	    int parameterNumber=1, G4bool reGet=true);
      G4double GetCurrentDoubleValue(const char * aCommand,
	    int parameterNumber=1, G4bool reGet=true);
      G4String GetCurrentStringValue(const char * aCommand, 
	    const char * aParameterName, G4bool reGet=true);
      G4int GetCurrentIntValue(const char * aCommand, 
	    const char * aParameterName, G4bool reGet=true);
      G4double GetCurrentDoubleValue(const char * aCommand,
	    const char * aParameterName, G4bool reGet=true);
      //  These six methods returns the current value of a parameter of the
      // given command. For the first three methods, the ordering number of
      // the parameter (1 is the first parameter) can be given, whereas,
      // other three methods can give the parameter name.
      //  If "reGet" is true, actual request of returning the current value
      // will be sent to the corresponding messenger, while, if it is false,
      // the value stored in G4Umanager will be used. The later case is valid
      // for the sequential invokation for the same command.

      inline void SetPauseAtBeginOfEvent(G4bool vl)
      { pauseAtBeginOfEvent = vl; }
      inline G4bool GetPauseAtBeginOfEvent() const
      { return pauseAtBeginOfEvent; }
      inline void SetPauseAtEndOfEvent(G4bool vl)
      { pauseAtEndOfEvent = vl; }
      inline G4bool GetPauseAtEndOfEvent() const
      { return pauseAtEndOfEvent; }
      //  If the boolean flags are true, Pause() method of G4StateManager is invoked
      // at the very begining (before generating a G4Event object) or at the end of
      // each event. So that, in case a (G)UI session is defined, the user can interact.


  public:
      inline G4UIcommandTree * GetTree() const
      { return treeTop; }
      inline G4UIsession * GetSession() const
      { return session; }
  public: // with description
      inline void SetSession(G4UIsession *const value)
      { session = value; }
      //  This method defines the active (G)UI session.
     void SetCoutDestination(G4UIsession *const value);
     //  This method defines the destination of G4cout/G4cerr stream.
     // For usual cases, this method will be invoked by a concrete
     // (G)UI session class object and thus the user needs not to invoke
     // this method by him(her)self.

  public:
      inline void SetVerboseLevel(G4int val)
      { verboseLevel = val; }
      inline G4int GetVerboseLevel() const
      { return verboseLevel; }
      inline G4int GetNumberOfHistory() const
      { return histVec.size(); }
      inline G4String GetPreviousCommand(G4int i) const
      { 
        G4String st;
        if(i>=0 && i<G4int(histVec.size()))
        { st = histVec[i]; }
        return st;
      }

  // Old methods kept for backward compatibility
  //    inline G4UIcommandTree * GetTree() const
  //    { return treeTop; };
  //    inline G4UIsession * GetSession() const
  //    { return session; };
  //    inline void SetSession(G4UIsession *const  value)
  //    { session = value; };

};

#endif

