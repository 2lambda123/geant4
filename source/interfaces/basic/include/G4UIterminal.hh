//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: G4UIterminal.hh,v 1.8 2003-06-16 16:55:57 gunter Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//

#ifndef G4UIterminal_h
#define G4UIterminal_h 1

#include <fstream>
#include "G4UImanager.hh"
#include "G4VBasicShell.hh"
#include "G4VUIshell.hh"

//
//   Description:
//
//   This class inherits the class G4UIsession.
//   This is the class to use a character-terminal sesion.
//
//   Usage:  
//       G4UIsession* terminalSession = new G4UIterminal; 
//   or  G4UIsession* terminalSession = new G4UIterminal(new your-shell); 
//
//     A character-terminal session  "terminalSession" is instantiated.
//     G4cout stream is redirected by default to the constructed instance.
//
//   terminalSession-> SessionStart(); // "terminalSession" is started.
//   delete terminalSession;           // "terminalSession"  is deleted.
//
//
//   In default(no arguments are given), csh-like shell is instantiated.
//   If you want to use another shell (e.g. tcsh-like), you can give
//   your favorite shell in an argument of the constructor.
//
//   Which shell? / How to define your own shell?
//   Currently two kinds of shells,
//                   G4UIcsh / G4UItcsh
//   , are presented. 
//   They inherit the abstract base class, G4VUIshell.
//   In order to define your own shell, 
//     - Define your own shell class derived from G4VUIshell.
//     - Implement GetCommandLine() method (pure virtual).
//     - Add more functionality, if need.
//
//   For more detail, see source codes.
//

class G4UIterminal : public G4VBasicShell {
private:
  G4UImanager* UI;
  // shell
  G4VUIshell* shell;

  // program states
  G4bool iExit;
  G4bool iCont;

public:
  G4UIterminal(G4VUIshell* aShell=0);
  ~G4UIterminal();

  void SetPrompt(const G4String& prompt);

private:
  void ExecuteCommand(G4String aCommand);
  G4String GetCommand(const char* msg=0);
  G4bool GetHelpChoice(G4int& aInt);
  void ExitHelp();
  
public:
  // These methods are implementation of corresponding virtual methods
  // of G4UIsession class.
  virtual G4UIsession* SessionStart();  
  virtual void PauseSessionStart(G4String msg);
  virtual G4int ReceiveG4cout(G4String coutString);
  virtual G4int ReceiveG4cerr(G4String cerrString);

};

#endif

