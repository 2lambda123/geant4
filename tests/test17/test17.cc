//
// ********************************************************************
// * DISCLAIMER                                                       *
// *                                                                  *
// * The following disclaimer summarizes all the specific disclaimers *
// * of contributors to this software. The specific disclaimers,which *
// * govern, are listed with their locations in:                      *
// *   http://cern.ch/geant4/license                                  *
// *                                                                  *
// * Neither the authors of this software system, nor their employing *
// * institutes,nor the agencies providing financial support for this *
// * work  make  any representation or  warranty, express or implied, *
// * regarding  this  software system or assume any liability for its *
// * use.                                                             *
// *                                                                  *
// * This  code  implementation is the  intellectual property  of the *
// * GEANT4 collaboration.                                            *
// * By copying,  distributing  or modifying the Program (or any work *
// * based  on  the Program)  you indicate  your  acceptance of  this *
// * statement, and all its terms.                                    *
// ********************************************************************
//
//
// $Id: test17.cc,v 1.10 2005-05-20 09:29:13 vnivanch Exp $
// GEANT4 tag $Name: not supported by cvs2svn $
//
// 
// --------------------------------------------------------------
//      GEANT 4 - test17
//
// --------------------------------------------------------------
// Comments
//
// 18.08.2000 V.Ivanchenko clean up visualisation and dummy output
//
// --------------------------------------------------------------

#include "G4RunManager.hh"
#include "G4UImanager.hh"
#include "G4UIterminal.hh"
#include "Randomize.hh"

#include "Test17DetectorConstruction.hh"
#include "Test17PhysicsList.hh"
#include "Test17PrimaryGeneratorAction.hh"
#include "Test17RunAction.hh"
#include "Test17EventAction.hh"
#include "Test17SteppingAction.hh"

int main(int argc,char** argv) {

  //choose the Random engine
  HepRandom::setTheEngine(new RanecuEngine);
  
  // Construct the default run manager
  G4RunManager * runManager = new G4RunManager;

  // set mandatory initialization classes
  Test17DetectorConstruction* detector;
  detector = new Test17DetectorConstruction;
  runManager->SetUserInitialization(detector);
  runManager->SetUserInitialization(new Test17PhysicsList(detector));

  // set user action classes
  runManager->SetUserAction(new Test17PrimaryGeneratorAction(detector));
  Test17RunAction* runaction = new Test17RunAction;
  runManager->SetUserAction(runaction);

  Test17EventAction* eventaction = new Test17EventAction(runaction);
  runManager->SetUserAction(eventaction);

  Test17SteppingAction* steppingaction = new Test17SteppingAction(detector,
                                               eventaction, runaction);
  runManager->SetUserAction(steppingaction);

  //Initialize G4 kernel
//  runManager->Initialize();

  // get the pointer to the User Interface manager
    G4UImanager* UI = G4UImanager::GetUIpointer();

  if (argc==1)   // Define UI terminal for interactive mode
    {
     G4UIsession * session = new G4UIterminal;
     UI->ApplyCommand("/control/execute init.mac");
     session->SessionStart();
     delete session;
    }
  else           // Batch mode
    {
     G4String command = "/control/execute ";
     G4String fileName = argv[1];
     UI->ApplyCommand(command+fileName);
    }

  // job termination
  delete runManager;

  return 0;
}

