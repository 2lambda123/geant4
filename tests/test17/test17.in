#
# Macro file for the initialization phase of "test17" based on "TestEm6.cc"
#
# It creates the default geometry (simple absorber box )        
#
/control/verbose 2
/event/setverbose 0
#
/run/initialize
#
/gun/particle proton
#/gun/zvertex -0.16 mum
/gun/energy 500 keV
/tracking/verbose 0
/run/beamOn 1000
#
/gun/particle anti_proton
/run/beamOn 1000

#scaled energy
/gun/particle IonC12
/gun/energy 6000 keV
/run/beamOn 1000

