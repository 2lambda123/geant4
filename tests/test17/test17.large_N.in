#
# Macro file for the initialization phase of "test17"
#
#
/control/verbose 2
/run/verbose 1
#
/run/initialize
#
/gun/particle proton
/gun/energy 500 keV
/tracking/verbose 0
/run/beamOn 10000
#
/gun/particle anti_proton
/run/beamOn 10000
#scaled energy
/gun/particle ion
/gun/ion 6 12 6
/gun/energy 6000 keV
/run/beamOn 10000
#scaled energy
/gun/particle ion
/gun/ion 26 56 26
/run/beamOn 1000
#scaled energy
/gun/particle ion
/gun/ion 6 12 6
/run/beamOn 1000
#scaled energy
/gun/particle mu+
/gun/energy   3.5 MeV
/run/beamOn 1000
