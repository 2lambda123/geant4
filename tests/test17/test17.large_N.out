Unit category <Error tolerance> is not defined.
**********************************************
 Geant4 version $Name: not supported by cvs2svn $
                                (29-Jun-2001)
             Copyright : Geant4 Collaboration
**********************************************
G4VUserPhysicsList::SetVerboseLevel  : Verbose level is set to 2
Matrial available: Beryllium
Matrial available: Aluminum
Matrial available: Carbon
Matrial available: Silicon
Matrial available: liquidArgon
Matrial available: Iron
Matrial available: Copper
Matrial available: Gold
Matrial available: Lead
Matrial available: Water
Matrial available: Methane
Matrial available: Graphite

***** Table : Nb of materials = 13 *****

 Material: Beryllium            density:  1.848 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  35.267 cm
   --->  Element: Beryllium    Z =  4.0   N =   9.0   A =   9.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Aluminum            density:  2.700 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   8.893 cm
   --->  Element: Aluminum    Z = 13.0   N =  27.0   A =  26.98 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Carbon            density:  2.000 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  21.348 cm
   --->  Element: Carbon    Z =  6.0   N =  12.0   A =  12.01 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Silicon            density:  2.330 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   9.368 cm
   --->  Element: Silicon    Z = 14.0   N =  28.1   A =  28.09 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: liquidArgon            density:  1.390 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  14.065 cm
   --->  Element: liquidArgon    Z = 18.0   N =  40.0   A =  39.95 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Iron            density:  7.870 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.759 cm
   --->  Element: Iron    Z = 26.0   N =  55.9   A =  55.85 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Copper            density:  8.960 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   1.436 cm
   --->  Element: Copper    Z = 29.0   N =  63.5   A =  63.55 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Gold            density: 19.320 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   3.344 mm
   --->  Element: Gold    Z = 79.0   N = 197.0   A = 196.97 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Lead            density: 11.350 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:   5.612 mm
   --->  Element: Lead    Z = 82.0   N = 207.2   A = 207.19 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Water        H_2O   density:  1.000 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  36.092 cm
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  11.21 %  Abundance  66.67 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  88.79 %  Abundance  33.33 %

 Material: Methane        CH_4   density:  0.667 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 696.375 m
   --->  Element: HydrogenH   Z =  1.0   N =   1.0   A =   1.01 g/mole  fractionMass:  25.19 %  Abundance  80.00 %
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass:  74.81 %  Abundance  20.00 %

 Material: Graphite        Graphite   density:  2.265 g/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength:  18.834 cm
   --->  Element: CarbonC   Z =  6.0   N =  12.0   A =  12.00 g/mole  fractionMass: 100.00 %  Abundance 100.00 %

 Material: Air            density:  1.000 mg/cm3  temperature: 273.15 K  pressure:   1.00 atm  RadLength: 367.858 m
   --->  Element: NitrogenN   Z =  7.0   N =  14.0   A =  14.01 g/mole  fractionMass:  70.00 %  Abundance  72.71 %
   --->  Element: OxygenO   Z =  8.0   N =  16.0   A =  16.00 g/mole  fractionMass:  30.00 %  Abundance  27.29 %



 The  WORLD   is made of 21 cm of Air. The transverse size (YZ) of the world is 12 cm
 The ABSORBER is made of 1 items of 10 cm of Water. The transverse size (YZ) is 10 cm
 X position of the middle of the absorber 0 fm
G4VUserPhysicsList::Construct()
Construct particles 
Construct processes 
Ionic processes for IonC12
Ionic processes for alpha
Hadronic processes for anti_proton
Ionic processes for deuteron
Hadronic processes for kaon+
Hadronic processes for kaon-
MuonMinusCaptureAtRest is created 
MuonMinusCaptureAtRest is created 
Hadronic processes for pi+
Hadronic processes for pi-
Hadronic processes for proton
Test17PhysicsList::SetCuts:CutLength : 100 mum
Set cuts for all particles! 
G4VUserPhysicsList::SetCutValue       :Set cuts for gamma

phot:  Total cross sections from a parametrisation. Good description from 10 KeV to 50 MeV for all Z
        Sandia crossSection below 50 KeV
       PhysicsTables from 50 keV to 50 MeV in 100 bins. 

compt:  Total cross sections from a parametrisation. Good description from 10 KeV to (100/Z) GeV. 
       Scattered gamma energy according Klein-Nishina.
        PhysicsTables from 10 keV to 100 GeV in 100 bins. 

conv:  Total cross sections from a parametrisation. Good description from 1.5 MeV to 100 GeV for all Z. 
        e+e- energies according Bethe-Heitler
        PhysicsTables from 1.022 MeV to 100 GeV in 100 bins. 
G4VUserPhysicsList::SetCutValue       :Set cuts for e-

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV to 100 TeV in 100 bins. 

eIoni:  delta cross sections from Moller+Bhabha. Good description from 1 KeV to 100 GeV.
        delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 100 TeV in 100 bins. 

eBrem:  Total cross sections from a NEW parametrisation based on the EEDL data library. 
 Good description from 1 KeV to 100 GeV.
        log scale extrapolation above 100 GeV 
        Gamma energy sampled from a parametrised formula.
        PhysicsTables from 1 keV to 100 TeV in 100 bins. 
G4VUserPhysicsList::SetCutValue       :Set cuts for e+

annihil:  Total cross section from Heilter formula(annihilation into 2 photons).
        gamma energies sampled according Heitler
        PhysicsTables from 10 keV to 10 TeV in 100 bins. 
G4VUserPhysicsList::SetCutValue       :Set cuts for mu-
G4VUserPhysicsList::SetCutValue       :Set cuts for mu+

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV to 100 TeV in 100 bins. 

MuIoni:    Knock-on electron cross sections . 
         Good description above the mean excitation energy.
         delta ray energy sampled from  differential Xsection.
        PhysicsTables from 1 keV to 1000 PeV in 150 bins. 

MuBrems:  theoretical cross section 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1000 PeV in 150 bins. 

MuPairProd:  theoretical cross sections 
          Good description up to 1000 PeV.
    PhysicsTables from 1 keV to 1000 PeV in 150 bins. 
G4VUserPhysicsList::SetCutValue       :Set cuts for proton

msc:   Tables of transport mean free paths.
          New model of MSC , computes the lateral 
          displacement of the particle , too.
        PhysicsTables from 100 eV to 100 TeV in 100 bins. 

hLowEIoni:    Knock-on electron cross sections . 
        Good description above the mean excitation energy.
        Delta ray energy sampled from  differential Xsection.
        PhysicsTables from 10 eV  to 100 TeV  in 200 bins.
        Electronic stopping power model is  ICRU_R49p
        from 1 keV  to 2 MeV .

        Parametrisation model for antiprotons is  ICRU_R49p
        from 5 keV  to 2 MeV .
        Parametrization of the Barkas effect is switched on.
        Nuclear stopping power model is ICRU_R49
G4VUserPhysicsList::SetCutValue       :Set cuts for anti_proton
G4VUserPhysicsList::SetCutValueForOthers   :0.1[mm]
Set cuts for He3
Set cuts for IonC12
Set cuts for alpha
Set cuts for anti_nu_e
Set cuts for anti_nu_mu
Set cuts for chargedgeantino
Set cuts for deuteron
Set cuts for geantino
Set cuts for kaon+
Set cuts for kaon-
Set cuts for neutron
Set cuts for nu_e
Set cuts for nu_mu
Set cuts for pi+
Set cuts for pi-
Set cuts for pi0
Set cuts for triton
============= The cut Energy ==============================
                      gamma            e-           
Cut in range                100 mum           100 mum
Cut in energy
 Beryllium                          984 eV           108 keV
   Aluminum                         2.33 keV           130 keV
   Carbon                         1.32 keV           118 keV
   Silicon                         2.29 keV           121 keV
   liquidArgon                         1.99 keV          82.5 keV
   Iron                         6.26 keV           235 keV
   Copper                         7.32 keV           250 keV
   Gold                         40.5 keV           348 keV
   Lead                         29.4 keV           241 keV
   Water                         1.08 keV          84.6 keV
   Methane                            1 eV          2.62 eV
   Graphite                         1.38 keV           128 keV
   Air                            1 eV          4.02 eV
===================================================
/run/verbose 1
#
#/run/initialize
#
/gun/particle proton
/gun/energy 500 keV
/tracking/verbose 0
/run/beamOn 10000
### Run 0 start.
Start Run processing.
EvtNo= 827  length= 0.00879862  xend= 0.00381338
EvtNo= 7065  length= 0.0186306  xend= -0.0118009
Run terminated.
Run Summary
  Number of events processed : 10000
  User=11.95s Real=12.05s Sys=0.01s
 ================== run summary =====================

 end of Run TotNbofEvents = 9998 for proton with Ekin = 0.5 MeV
    Track Length in absorber = 9.21897 +- 0.00208969  microns 
    CSDA  Range  in absorber = 9.00586 +- 0.00212364  microns 

    Energy deposit in absorber = 0.5  MeV 

### Comparison with the ICRU49 data: 
    Track Length (G4 - ICRU49) = 0.159972 +- 0.00208969 microns 
    CSDA  Range  (G4 - ICRU49) = 0.13686 +- 0.00212364 microns 

Average Number electrons per event = 0
Average Number photons per event   = 0

#
/gun/particle anti_proton
/run/beamOn 10000
### Run 1 start.
Start Run processing.
EvtNo= 160  length= 0.00962096  xend= 0.00214699
EvtNo= 341  length= 0.0095304  xend= 0.00412484
EvtNo= 1807  length= 0.00950738  xend= 0.00339889
EvtNo= 4410  length= 0.00936745  xend= 0.00372834
EvtNo= 5686  length= 0.00936745  xend= 0.00286315
EvtNo= 8924  length= 0.00935399  xend= 0.00410856
EvtNo= 8928  length= 0.00936745  xend= 0.00415206
Run terminated.
Run Summary
  Number of events processed : 10000
  User=13.77s Real=14.04s Sys=0s
 ================== run summary =====================

 end of Run TotNbofEvents = 9993 for anti_proton with Ekin = 0.5 MeV
    Track Length in absorber = 9.58539 +- 0.00215436  microns 
    CSDA  Range  in absorber = 9.27129 +- 0.0027553  microns 

    Energy deposit in absorber = 0.5  MeV 

Average Number electrons per event = 0
Average Number photons per event   = 0

#scaled energy
/gun/particle IonC12
/gun/energy 6000 keV
/run/beamOn 10000
### Run 2 start.
Start Run processing.
EvtNo= 4926  length= 0.00841309  xend= 0.00403149
Run terminated.
Run Summary
  Number of events processed : 10000
  User=13.54s Real=13.86s Sys=0.03s
 ================== run summary =====================

 end of Run TotNbofEvents = 9999 for IonC12 with Ekin = 6 MeV
    Track Length in absorber = 8.57201 +- 0.00148487  microns 
    CSDA  Range  in absorber = 7.9612 +- 0.00478038  microns 

    Energy deposit in absorber = 6  MeV 

Average Number electrons per event = 0
Average Number photons per event   = 0

G4 kernel has come to Quit state.
